meta-mender-raspberrypi/recipes-bsp/u-boot/patches/0001-CONFIGS-rpi-enable-mender-requirements.patch

```diff
@@ -11,73 +11,131 @@ Mender also requires that the environment is on MMC
 Signed-off-by: Mirza Krak <mirza.krak@gmail.com>
 Signed-off-by: Drew Moseley <drew.moseley@northern.tech>
 ---
- configs/rpi_0_w_defconfig   | 1 +
- configs/rpi_2_defconfig     | 1 +
- configs/rpi_3_32b_defconfig | 1 +
- configs/rpi_3_defconfig     | 1 +
- configs/rpi_defconfig       | 1 +
+ configs/rpi_0_w_defconfig   | 3 +--
+ configs/rpi_2_defconfig     | 3 +--
+ configs/rpi_3_32b_defconfig | 3 +--
+ configs/rpi_3_defconfig     | 3 +--
+ configs/rpi_defconfig       | 3 +--
+ env/Kconfig                 | 1 -
  include/configs/rpi.h       | 3 +++
- 6 files changed, 8 insertions(+)
+ 7 files changed, 8 insertions(+), 11 deletions(-)
 
 diff --git a/configs/rpi_0_w_defconfig b/configs/rpi_0_w_defconfig
-index 9a6d24be22..33e0c86e2a 100644
+index 39da54c3e7..9a0fc1f0b5 100644
 --- a/configs/rpi_0_w_defconfig
 +++ b/configs/rpi_0_w_defconfig
-@@ -25,3 +25,4 @@ CONFIG_SYS_WHITE_ON_BLACK=y
+@@ -16,8 +16,6 @@ CONFIG_CMD_USB=y
+ CONFIG_CMD_FS_UUID=y
+ CONFIG_OF_EMBED=y
+ CONFIG_DEFAULT_DEVICE_TREE="bcm2835-rpi-zero-w"
+-CONFIG_ENV_FAT_INTERFACE="mmc"
+-CONFIG_ENV_FAT_DEVICE_AND_PART="0:1"
```

Comment:
Instinctively, this looks like a wrong change to me, but maybe it has been moved somewhere else? This should probably be tested on hardware to be sure, especially that `fw_setenv` works and is reflected at the boot prompt.

Reply:
Above will try to load the U-boot environment from an file on the fat partition, which is not our intention.

This has always been there, but on previous U-boot versions the config `CONFIG_ENV_IS_IN_MMC=y` which we set seems to have taken precedence, but on this U-boot version it tried using FAT if this was not removed. There is a another change further down in this patch which also removes,

```
config ENV_IS_IN_FAT
...
default y if ARCH_BCM283X
```

Which is new for this U-boot version as well.

Reply:
You're right, easy to misread those defines. Disregard my comment then.